support-frontend/assets/pages/paper-subscription-checkout/helpers/homeDeliveryDays.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+import type { ProductOptions } from 'helpers/productPrice/productOptions';
+import type { Day } from 'helpers/subscriptionsForms/deliveryDays';
+import {
+  getDeliveryDays,
+  jsDayToFulfilmentDay,
+  numberOfWeeksWeDeliverTo,
+} from 'helpers/subscriptionsForms/deliveryDays';
+
+const everyDayAndSixDay = [3, 3, 3, 3, 6, 5, 4];
+const weekendAndSaturday = [6, 5, 4, 3, 9, 8, 7];
+const sunday = [7, 6, 5, 4, 10, 9, 8];
+
+const getDaysToAdd = (today: number, product: ProductOptions): number => {
+  switch (product) {
+    case 'Everyday':
+    case 'EverydayPlus':
+    case 'Sixday':
+    case 'SixdayPlus':
+      return everyDayAndSixDay[today];
+    case 'Weekend':
+    case 'WeekendPlus':
+    case 'Saturday':
+    case 'SaturdayPlus':
+      return weekendAndSaturday[today];
+    default:
+      return sunday[today];
+  }
+};
+
+const daysTillNextDelivery = (
+  currentWeekDay: number,
+  deliveryDay: number,
+): number =>
+  (currentWeekDay > deliveryDay ?
+    (7 - currentWeekDay) + deliveryDay :
+    deliveryDay - currentWeekDay);
+
+const canDeliverOnNextDeliveryDay = (
+  currentWeekDay: number,
+  deliveryDay: number,
+  delayDays: number,
+): boolean => daysTillNextDelivery(currentWeekDay, deliveryDay) >= delayDays;
+
+const getHomeDeliveryDays = (today: number, product: ProductOptions): Date[] => {
+  const currentWeekday = new Date(today).getDay();
+  const delayDays = getDaysToAdd(currentWeekday, product);
+  const deliveryDay: Day = ((currentWeekday + delayDays) % 7: any);
+  const canMakeNextDelivery = canDeliverOnNextDeliveryDay(currentWeekday, deliveryDay, delayDays);
```

Comment:
Am I missing something or could we just do `const firstOption = today.addDays(delayDays)` for the first delivery option and then `daysTillNextDelivery(firstOption)` recursively for further dates?

Reply:
I wanted to use the `getDeliveryDays()` function that all the other products including voucher use for consistency's sake, which means that we need to give it the delivery day of the week and then work out whether the first date it returns us is valid or not ie. whether we can fulfil by that date in that week.